Tasks/DeployAzureResourceGroup/AzureResourceManagerHelper.ps1

```diff
@@ -170,40 +170,47 @@ function Get-MachineConnectionInformation
 		$publicIPAddressResources = Get-AzurePublicIpAddress -ResourceGroupName $resourceGroupName -ErrorAction Stop -Verbose
 		Set-Variable -Name publicIPAddressResources -Value $publicIPAddressResources -Scope "Global"
 
-		$lb = $azureResourceGroup.Resources |  Where-Object {$_.ResourceType -eq "Microsoft.Network/loadBalancers"}
-
-		if($lb.Count -gt 1)
-		{
-			Throw (Get-LocalizedString -Key "Current version of the task supports only templates with single load balancer.")
-		}
-		
-		if($lb)
-		{
-			$loadBalancer = Get-AzureLoadBalancer -Name $lb.Name -ResourceGroupName $resourceGroupName -ErrorAction Stop -Verbose
-			Set-Variable -Name loadBalancer -Value $loadBalancer -Scope "Global"
-
-			$fqdnMap = Get-MachinesFqdnsForLB -resourceGroupName $resourceGroupName
-			$winRmHttpPortMap = Get-FrontEndPorts -BackEndPort "5985"
-			$winRmHttpsPortMap = Get-FrontEndPorts -BackEndPort "5986"
-		}
-		else
+		$lbGroup = $azureResourceGroup.Resources |  Where-Object {$_.ResourceType -eq "Microsoft.Network/loadBalancers"}
+        
+        $fqdnMap = @{}
+        Set-Variable -Name fqdnMap -Value $fqdnMap -Scope "Global"
+
+        $winRmHttpPortMap = @{}
+        Set-Variable -Name winRmHttpPortMap -Value $winRmHttpPortMap -Scope "Global"
+
+        $winRmHttpsPortMap = @{}
+        Set-Variable -Name winRmHttpsPortMap -Value $winRmHttpsPortMap -Scope "Global"
+        
+        if($lbGroup.Count -gt 0)
+        {
+            foreach($lb in $lbGroup)
+            {
+                $loadBalancer = Get-AzureLoadBalancer -Name $lb.Name -ResourceGroupName $resourceGroupName -ErrorAction Stop -Verbose
+			    Set-Variable -Name loadBalancer -Value $loadBalancer -Scope "Global"
+
+			    $fqdnMap = Get-MachinesFqdnsForLB -resourceGroupName $resourceGroupName
```

Comment:
have you checked that all code here is re-entrant and safe? and for below methods as well?